src/classes/storage.ts

```diff
@@ -84,4 +100,155 @@ export class Storage {
+      groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.STOP_GROUP),
+    );
+
+    const filteredByCluster: AutoComplete[] = this.filterGroundPlacesByCluster(
+      groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.STOP_CLUSTER),
+    );
+
+    const filteredByType: AutoComplete[] = [...filteredByGroup, ...filteredByCluster];
+
+    const filteredBySegmentProviderStop: AutoComplete[] = this.filterGroundPlacesBySegmentProvider(
+      // Filtering by segment provider must be made on the whole results previously filtered.
+      // We checks here if this filter must be applied on results already filtered or not.
+      filteredByType.length ? filteredByType : groundPlacesAutocomplete,
+      filters.includes(AutoCompleteFilters.SEGMENT_PROVIDER_STOP),
+    );
+
+    const filteredByService: AutoComplete[] = this.filterGroundPlacesByServiced(
+      filteredBySegmentProviderStop,
+      filters.includes(AutoCompleteFilters.SERVICED),
+    );
+
+    return filteredByService;
+  }
+
+  /**
+   * @description Search through ground places with specific words.
+   * @param {AutoComplete[]} groundPlaces - The places on which the search is used.
+   * @param {string} search - The search to find the requested places.
+   * @returns {AutoComplete[]}
+   */
+  public searchGroundPlaces(groundPlaces: AutoComplete[], search: string): AutoComplete[] {
+    return groundPlaces.filter((place) => {
+      // Method toLowerCase() is used because the includes() method is case sensitive
+      const currentSearch = search.toLowerCase();
+
+      /* Search by unique name. This search only concern StopCluster. */
+      if (place.type === GroundPlaceType.CLUSTER && place.unique_name.toLowerCase().includes(currentSearch)) {
+        return place;
+      }
+
+      /* Search by Gpuid and by name */
+      if (place.gpuid.toLowerCase().includes(currentSearch) || place.name.toLowerCase().includes(currentSearch)) {
+        return place;
+      }
+    });
+  }
+
+  // -----------------------------------------------------------------------------
+  // ----------------------------- PRIVATE METHODS -------------------------------
+  // -----------------------------------------------------------------------------
+
+  /**
+   * @description Converts the Ground places to an array list manipulable.
+   * @returns {AutoComplete[]}
+   */
+  private parseGroundPlaces(): AutoComplete[] {
+    return Object.entries(this.groundPlaces).map(
+      ([gpuid, place]: [StopGroupGpuid | StopClusterGpuid, StopGroup | StopCluster]) => ({
+        gpuid,
+        ...place,
+      }),
+    );
+  }
+
+  /**
+   * @description Filter Ground places by group type.
+   * @param {AutoComplete[]} groundPlaces - Ground places list to be filtered.
+   * @param {boolean} filterByGroup - Filter or not by group type.
+   * @returns {AutoComplete[]}
+   */
+  private filterGroundPlacesByGroup(groundPlaces: AutoComplete[], filterByGroup: boolean): AutoComplete[] {
```

Comment:
Why one method per filter ? 
You're method all do the same thing.

Reply:
I chose this way to make the filters works in order to avoid ending up with a big block that is difficult to maintain. Each filter is used on a specific element of the place (it can a StopGroup or a StopCluster). They can be used alone or combined.

Reply:
They share the same structure (for three of them), but not the same goal.

Reply:
Super complicated. 100% disagree. 
One function is simpler to maintain than three

Reply:
No need for if inside. `filter`method exists.